maths/nth_sgonal_num.py

```diff
@@ -0,0 +1,21 @@
+def nth_sgonal_num(n: int, s: int) -> int:
+    """
```

Comment:
Improve variable naming: The variables n and s are not very descriptive. Consider `n` as `num` and `s` as `side`.